### ISSUE TITLE
--nn, --audit and flatbuffers generate different results

### ISSUE THREAD

**Issue by lalo** [VowpalWabbit/vowpal_wabbit#2803]
https://github.com/VowpalWabbit/vowpal_wabbit/blob/cb37da4553c475b4d563e22f43194adc8f127cd2/test/run_tests.py#L540

The test is currently disabled, but it is failing. Not sure if --audit is a factor here or --nn, it could have a more minimal repro.

Previous diff failure before being disabled:

```Test 270: Fail
	Description: vw test used in brew script
	vw_command: "-d /home/vsts_azpcontainer/.vw_fb_runtests_working_dir/test_270/houses.fb --audit --nn 1 --flatbuffer"
	[stdout] Fail: Diff not OK, Lines differ - ref vs output: '-0.017576 third_house' vs '-0.517576 third_house'

--- train-sets/ref/houses.stdout
+++ stdout
@@ -20,11 +20,11 @@
 OutputWeight:156908:1:-0.302343@0.815926
 -0.316100
 HiddenBias:29015:1:-0.3161@0.99997
--0.017576 third_house
+-0.517576 third_house
 price:197660:0.53:-0.335796@0.0236938	age:68258:0.87:-0.172693@0.089583	sqft:63562:0.32:-0.402875@0.016454	Constant:232120:1:-0.0604413@0.73129	1924:3848:1:0@0
 -0.302343
 OutputWeight:156908:1:-0.302343@0.815926
-1
-OutputLayerConst:156909:1:0.506834@8	OutputLayer0:156908:-0.017558:-0.302343@0.815926
--0.290302
-OutputWeight:156908:1:-0.290302@0.817159
+0.650698
+OutputLayerConst:156909:1:0.506834@8	OutputLayer0:156908:-0.475829:-0.302343@0.815926
+-0.141233
+OutputWeight:156908:1:-0.141233@1.19939

	[stderr] Fail: Diff not OK, Lines differ - ref vs output: 'average loss = 0.750000' vs 'average loss = 0.605852'

--- /home/vsts_azpcontainer/.vw_fb_runtests_working_dir/test_270/houses.stderr
+++ stderr
@@ -15,7 +15,7 @@
 number of examples = 3
 weighted example sum = 4.000000
 weighted label sum = 2.000000
-average loss = 0.750000
+average loss = 0.605852
 best constant = 0.500000
 best constant's loss = 0.250000
 total feature number = 15

**Comment by lalo** [comment]
Test got added bc of this issue: https://github.com/VowpalWabbit/vowpal_wabbit/issues/2799
